src/map/script.c

```diff
@@ -17784,14 +17784,18 @@ static BUILDIN(getd)
 	char varname[100];
 	const char *buffer;
 	int elem;
+	int id;
 
 	buffer = script_getstr(st, 2);
 
 	if (sscanf(buffer, "%99[^[][%d]", varname, &elem) < 2)
 		elem = 0;
 
+	id = script->add_str(varname);
```

Comment:
what if varname already added with other type?
assign C_NAME will break other things.

Probably better check is varname not added yet, and if it not added, add it and set type to C_NAME?